link_test.go

```diff
@@ -2591,14 +2591,32 @@ func TestLinkXdp(t *testing.T) {
 	if err != nil {
 		t.Skipf("Loading bpf program failed: %s", err)
 	}
+	t.Cleanup(func() {
+		_ = unix.Close(fd)
+	})
+
 	if err := LinkSetXdpFd(testXdpLink, fd); err != nil {
 		t.Fatal(err)
 	}
-	if err := LinkSetXdpFdWithFlags(testXdpLink, fd, nl.XDP_FLAGS_UPDATE_IF_NOEXIST); !errors.Is(err, unix.EBUSY) {
-		t.Fatal(err)
+	t.Cleanup(func() {
+		_ = LinkSetXdpFd(testXdpLink, -1)
+	})
+	var err2 error
+	// It can take a moment for the kernel to update the link state, so we retry here
+	attempts := 0
+	for attempts = 0; attempts < 10; attempts++ {
```

Comment:
Can you please `for ; attempts < 10; attempts++ {`

Reply:
done